ImageD11/sinograms/dataset.py

```diff
@@ -446,7 +517,13 @@ def load( self, h5name, h5group = '/' ):
 
 
 def load( h5name, h5group = '/' ):
-    return DataSet().load( h5name, h5group )
+    # return DataSet().load( h5name, h5group )
+    # make sure detector and rotation motors are known
+    ds_obj = DataSet().load( h5name, h5group )
+    ds_obj.guess_detector()
```

Comment:
Silently modifying the dataset when loading it from disk seems a bit dodgy. I prefer to have a method like `validate` or `repair` or `fix` that makes clear what is happening.